prowler/lib/outputs/jira/jira.py

```diff
@@ -0,0 +1,1156 @@
+        - _cloud_id: The cloud ID
+        - _scopes: The scopes needed to authenticate, read:jira-user read:jira-work write:jira-work
+        - AUTH_URL: The URL to authenticate with Jira
+        - PARAMS_TEMPLATE: The template for the parameters to authenticate with Jira
+        - TOKEN_URL: The URL to get the access token from Jira
+        - API_TOKEN_URL: The URL to get the accessible resources from Jira
+
+    Methods:
+        - __init__: Initialize the Jira object
+        - input_authorization_code: Input the authorization code
+        - auth_code_url: Generate the URL to authorize the application
+        - get_auth: Get the access token and refresh token
+        - get_cloud_id: Get the cloud ID from Jira
+        - get_access_token: Get the access token
+        - refresh_access_token: Refresh the access token from Jira
+        - test_connection: Test the connection to Jira and return a Connection object
+        - get_projects: Get the projects from Jira
+        - get_available_issue_types: Get the available issue types for a project
+        - send_findings: Send the findings to Jira and create an issue
+
+    Raises:
+        - JiraGetAuthResponseError: Failed to get the access token and refresh token
+        - JiraGetCloudIDNoResourcesError: No resources were found in Jira when getting the cloud id
+        - JiraGetCloudIDResponseError: Failed to get the cloud ID, response code did not match 200
+        - JiraGetCloudIDError: Failed to get the cloud ID from Jira
+        - JiraAuthenticationError: Failed to authenticate
+        - JiraRefreshTokenError: Failed to refresh the access token
+        - JiraRefreshTokenResponseError: Failed to refresh the access token, response code did not match 200
+        - JiraGetAccessTokenError: Failed to get the access token
+        - JiraNoProjectsError: No projects found in Jira
+        - JiraGetProjectsError: Failed to get projects from Jira
+        - JiraGetProjectsResponseError: Failed to get projects from Jira, response code did not match 200
+        - JiraInvalidIssueTypeError: The issue type is invalid
+        - JiraGetAvailableIssueTypesError: Failed to get available issue types from Jira
+        - JiraGetAvailableIssueTypesResponseError: Failed to get available issue types from Jira, response code did not match 200
+        - JiraCreateIssueError: Failed to create an issue in Jira
+        - JiraSendFindingsResponseError: Failed to send the findings to Jira
+        - JiraTestConnectionError: Failed to test the connection
+
+    Usage:
+        jira = Jira(
+            redirect_uri="http://localhost:8080",
+            client_id="client_id",
+            client_secret="client_secret
+        )
+        jira.send_findings(findings=findings, project_key="KEY")
+    """
+
+    _redirect_uri: str = None
+    _client_id: str = None
+    _client_secret: str = None
+    _access_token: str = None
+    _refresh_token: str = None
+    _auth_expiration: int = None
```

Comment:
Is this an `int` or `datetime`? Maybe `_expiry_date` is better.

Reply:
Done